### ISSUE TITLE
Issue in converting in windows

### ISSUE THREAD

**Issue by saurav12994** [dhteodoro/sus-openehr-builder#1]
The input data should be stored in a relational database and a file with the patient ids should be provided as input.
Patient ids for the whole database can be found in two files:

apac_cnspcn.txt for the outpatient patients
aih_n_aih.txt for the hospitalized patients
For subsets of the database, the two patient ids files must be generated by extracting the corresponding  subsets of the full id files.

The database connection parameters should be set in the builder.properties

How can we done all step in windows

**Comment by sergiofreire** [comment]
Apart from changing the paths to the files I think that if you follow the instructions it will work fine.

Do you have any specific doubts?

**Comment by saurav12994** [comment]
I am converting the zip filer to jar file using https://convertio.co/zip-jar 
Then  it showing invalid or corrupt jar file in windows

**Comment by sergiofreire** [comment]
But you should not convert the zip files to jar. You must extract them
somewhere!

**Comment by saurav12994** [comment]
java -Xmx8g -Dfile.encoding=UTF-8 -jar “sus-openeher-builder.jar” –type ehr –format xml –patients ./patientIds/apac_cnspcn.txt –ehr-dir ./ehr
But this want jar file  i download zip file from github and then extract it and edit the builder.properties  and again converted it to zip file and then convert it using online tools
Please tell where i went  wrong

**Comment by sergiofreire** [comment]
Go to the folder where you extracted the zip file (it will be your
projectdir), edit the builder.properties and extract the terminology.tar.gz
file in “project-dir/repository/terminologies.

Then in the projectdir folder execute the command below:

java -Xmx8g -Dfile.encoding=UTF-8 -cp
projectdir/bin/uber-sus-openehr-builder-1.0.1-SNAPSHOT.jar
br.uerj.lampada.openehr.susbuilder.EHRGenerator --patients patientids
--ehr-dir outputdir --type < ehr|version|composition > --format < json|xml
> [--aih]

It will take a lot of time if you are using the full patientIds files. Just
to get a feeling of what is generated, extract a sample of the ids in the
patientId file and run the program with it.

**Comment by saurav12994** [comment]
![image](https://user-images.githubusercontent.com/27879922/27921327-105917c6-6295-11e7-8cb6-62cf7f859572.png)

**Comment by sergiofreire** [comment]
you must run the program from the "sus-openehr-builder-master" folder.

**Comment by saurav12994** [comment]
i am running from sus-openehr-builder-master folder i think there is some error in my builder file
Given below is my builder fileand my pgadminscreenshot please check 
# PostgreSQL
db.url=jdbc:postgresql://localhost:5432/orbda 
db.username="postgres" 
db.password="9914103044"

pretty.print=false
log.step=10000

archetype.repository=./repository/archetypes
template.repository=./repository/templates
terminology.file=./repository/terminologies/terminology.csv

log.config=log4j.properties

![image](https://user-images.githubusercontent.com/27879922/27921749-66b79286-6296-11e7-8719-1d8223a56c77.png)

**Comment by sergiofreire** [comment]
Try removing the quotation marks from the user and password.

**Comment by saurav12994** [comment]
still i getting same error

**Comment by saurav12994** [comment]
please help

**Comment by sergiofreire** [comment]
I downloaded the project from github, edited the builder.properties,
extracted the terminology.csv file and ran the program. It worked.
I get your error when I have errors in the connection parameters to
postgres.

I do not have a windows machine to try it out in windows for the time
being.

**Comment by saurav12994** [comment]
![image](https://user-images.githubusercontent.com/27879922/27930406-0c8fa23a-62b4-11e7-8ab8-5b8cc9751e6a.png)
i got error in line no 138 please check 
i try many way to solve error but can't resolve it

**Comment by saurav12994** [comment]
i think there is some error in my url  but i check syntax of sample url , then my url is same as sample url 
please help

**Comment by sergiofreire** [comment]
I tested the program in a windows machine and it ran without problems.

**Comment by saurav12994** [comment]
please explain step what you son in window machine 
My step-
1- install postgresql
2- create database orbda
3- create  all 7 table
4- insert data in all table 
5- insert  apac_cnspcn.txt  ,aih_n_aih.txt in sus-openehr-builder-master Folder 
6. Extract  sus-openehr-builder-master\repository\terminologies\terminology.tar to same location where terminology.tar present
7.following changes  in builder in sus-openehr-builder-master Folder 
PostgreSQL
db.url=jdbc:postgresql://localhost:5432/orbda 
db.username=postgres 
db.password=saurav12994
8, Add following path in system variable in envionment variable 
C:\Program Files\PostgreSQL\9.5\lib
C:\Program Files\PostgreSQL\9.5\lib
9. make folder named "oo" in sus-openehr-builder-master Folder  for output 
10. execute this query 
java -Xmx8g -Dfile.encoding=UTF-8 -cp G:\Major2\sus-openehr-builder-master\bin\uber-sus-openehr-builder-1.0.1-SNAPSHOT.jar br.uerj.lampada.openehr.susbuilder.EHRGenerator  -type ehr -format xml -patients apac_cnspcn.txt -ehr-dir G:\Major2\sus-openehr-builder-master\oo 
Please check step and tell me what step am i missing? or which step is wrong?

**Comment by sergiofreire** [comment]
I don't use step 8. The first 7 steps are correct.
Have you tried to debug the program and see what are the variables' values
just before the excepction is raised?

I will be away for a few days without any internet connection.

**Comment by saurav12994** [comment]
yes, I think there was some error in connection setup using  DbHandler and if condition is showing error in CompositionManager.java line no 138 , I also tried to manually put value of username ,url and password in compositionmanager.java  to run program but it still get same error

**Comment by saurav12994** [comment]
can you give me query that you executed in window machine please?

**Comment by saurav12994** [comment]
I installed the jdbc from application stack builder (which was installed in my system after installation of postgresSql) Do i need to install any thing for this project ?
![image](https://user-images.githubusercontent.com/27879922/28307556-e3a992fc-6bc0-11e7-9f12-f2f3115d94e9.png)

**Comment by sergiofreire** [comment]
The postgresql jdbc driver already comes with the source code in github. No
need to add others. I have not added the path in your step 8 to the system
variable either.

**Comment by sergiofreire** [comment]
On Sat, Jul 15, 2017 at 5:54 AM, saurav12994 <notifications@github.com>
wrote:

> can you give me query that you executed in window machine please?
>
What do you mean?

**Comment by sergiofreire** [comment]
Sorry, but I cannot see what is wrong with DbHandler. The code was written
by Douglas Teodoro, not me, but it is a typical code for connecting to
databases and we tested it in several machines over the last two years.
Anyway you can replace it and use your own code to connect to the database.

**Comment by dhteodoro** [comment]
Please, check whether the Postgres installation version you have is compatible with driver 9.2-1003-jdbc4

**Comment by saurav12994** [comment]
i have installed postgresql 9.5

**Comment by saurav12994** [comment]
when i am trying  to change some code in .java file it still showing same error .
for ex. i added  system.out.println("hello world ")  above the line showing error   then it showing error in same line no. it doesn't printing hello world in output
2nd time when i move  all java file from folder sus-openehr-builder-master\src\main\java\br\uerj\lampada\openehr\susbuilder to anywhere than i execute program , then it is showing same error even after i moved java files to other location. 
whatever i do with java file ,it doesn;t affecting my output
why?
plz help

**Comment by adityabangia** [comment]
@saurav12994 were you able to fix the error?
